fdb-record-layer-lucene/docs/index.md

```diff
@@ -2,14 +2,141 @@
+[indexspace][0, t1, pk1] -> partition metadata for records greater than t0 (the primary key is added to deduplicate if needed)
+    ⋮
+[indexspace][0, t1, pk2] -> partition metadata for "most recent" partition
+[indexspace][1, 0] -> the data for the directory in the 0th partition 
+[indexspace][1, 1]
+    ⋮
+[indexspace][1, 13]
+```
+
+#### grouped, partitioned
+If both grouping and partitioning are enabled, the tuples are concatenated.
+```
+[indexspace][g0, 0, t1, pk0]
+[indexspace][g0, 0, t2, pk1]
+    ⋮
+[indexspace][g0, 0, t3, pk2]
+[indexspace][g0, 1, 0]
+[indexspace][g0, 1, 1]
+    ⋮
+[indexspace][g0, 1, 13]
+[indexspace][g0, 0, t0, pk3]
+[indexspace][g0, 0, t1, pk4]
+    ⋮
+[indexspace][g0, 0, t2, pk5]
+[indexspace][g0, 1, 0]
+[indexspace][g0, 1, 1]
+    ⋮
+[indexspace][g0, 1, 13]
+```
+
+### Within a Directory
+
+
+#### Sequence Space
+```
+[directory-subspace][0] -> [Atomic Integer]
+```
+This stores a counter that is used for mapping various data below to an integer.
+
+#### Metadata Space
+```
+[directory-subspace][1][filename1] -> {id=1, size=234, ..., content=0xabcde234098}
+[directory-subspace][1][filename2] -> {id=1, size=234, ...}
+    ⋮
+```
+Lucene is designed to store a bunch of files, this maps file names to an id (created via the sequence number), 
+It also stores metadata about the files, such as size, and block size. See `FDBLuceneFileReference`.
+For some files, that are always very small, and always read, we store the content here directly, and not in the data 
+space (see below).
+The key is made up of the fixed `[1]`, followed by a tuple containing the filename.
+
+#### Data Space
+```
+[directory-subspace][2][1, 0] -> file content
+[directory-subspace][2][1, 1] -> file content
+[directory-subspace][2][2, 0] -> file content
+[directory-subspace][2][2, 1] -> file content
+[directory-subspace][2][2, 2] -> file content
+    ⋮
+```
+Each file is broken into a bunch of blocks, of a fixed size.
+The key is made up of the fixed `[2]`, followed by a tuple containing the file id (from the file reference in the
+Metadata subspace), and a block number. Each block is a fixed  size.
+
+#### Schema subspace
+A deprecated subspace of `[3]`
+
+#### PrimaryKey Subspace
+```
+[directory-subspace][4][pk0][1, 13]
+[directory-subspace][4][pk0][2, 33]
+[directory-subspace][4][pk1][1, 8]
+    ⋮
+```
+If enabled (via the index options `primaryKeySegmentIndexEnabled` (deprecated) or `primaryKeySegmentIndexV2Enabled`), this stores a
+mapping from the records primary key to the segment(s) that have that document. This is used for efficient deletes.
+The key is made up of the fixed `[4]`, followed by the primary key, followed by a tuple containing the `segmentId` and
+the `docId` within that segment.
+The `segmentId` is derived by creating a file reference for a fake `.pky` file for the segment.
+Note: A record may exist in multiple segments if the segment is in the process of being merged, or a merge failed part
```

Comment:
Maybe worthwhile to add that the information in this subspace will reflect the latest (most up to date) info about the record's location